tests/ui/delegation/glob-non-fn.rs

```diff
@@ -0,0 +1,38 @@
+#![feature(fn_delegation)]
+#![allow(incomplete_features)]
+
+trait Trait {
+    fn method(&self);
+    const CONST: u8;
+    type Type;
+    #[allow(non_camel_case_types)]
+    type method;
+}
+
+impl Trait for u8 {
+    fn method(&self) {}
+    const CONST: u8 = 0;
+    type Type = u8;
+    type method = u8;
+}
+
+struct Good(u8);
+impl Trait for Good {
+    reuse Trait::* { &self.0 }
+    // Explicit definitions for non-delegatable items.
+    const CONST: u8 = 0;
+    type Type = u8;
+    type method = u8;
+}
+
+struct Bad(u8);
+impl Trait for Bad { //~ ERROR not all trait items implemented, missing: `CONST`, `Type`, `method`
+    reuse Trait::* { &self.0 }
```

Comment:
(non-blocking) What do you think about skipping any non-fns? Not sure if that'd be less or more confusing. The diagnostics right now are certainly not helping ^^

Reply:
The plan is to eventually support delegating non-function associated items too.
For now they are just supposed to report some kind of error (which they do).
I agree that the current error is suboptimal, but that's not very critical.